src/storage-client/src/controller/persist_handles.rs

```diff
@@ -150,6 +181,16 @@ impl<T: Timestamp + Lattice + Codec64> PersistReadWorker<T> {
         self.send(PersistReadWorkerCmd::Downgrade(frontiers))
     }
 
+    pub(crate) fn snapshot_stats(
+        &self,
+        id: GlobalId,
+        as_of: Antichain<T>,
+    ) -> oneshot::Receiver<SnapshotStatsRes<T>> {
+        let (tx, rx) = oneshot::channel();
+        self.send(PersistReadWorkerCmd::SnapshotStats(id, as_of, tx));
+        rx
+    }
```

Comment:
Can you just make this `async` and `rx.await` at the end? Actually, you can do `rx.await.await` I think? all uses of this immediately double-await right?

Reply:
yeah, I have the same question. I was matching the couple of append methods on PersistWriterWorker that allow the caller to wait on the response, which do this pattern of non-async fn that returns a oneshot::Receiver. wasn't sure how intentional/reasoned that was

Reply:
We do the `oneshot` pattern for when we want to run a command and we both don't care about its result and we don't want the caller to be responsible for driving the future to completion. I think all the commands change the state in some way or another.

The command introduced in this PR doesn't really fit this pattern, it's just reading data, right? Why can't the controller just call `snapshot_stats()` on a handle directly when it needs to? If it's a read operation it never makes sense to not care about the result since there are no side effects.

So my suggestion in order of preference would be:
1. Don't make this a worker command and instead do the async work directly in `StorageController::snapshot_stats`
2. Leave it as a worker command but await the async work in the worker so that the oneshot contains the final result instead of a nested `Future`, maybe with a TODO comment for a future refactor.

Reply:
which handle would it use? the only ones I saw lying around were inside PersistReadWorker and i'm not sure how it would get at them besides going through the worker command pattern. we don't want to register a throwaway handle just for this because it'll be called once per dataflow input per SELECT/CREATE MATERIALIZED VIEW